lib/class-wp-theme-json-gutenberg.php

```diff
@@ -3247,6 +3258,27 @@ public function merge( $incoming ) {
 				}
 			}
 		}
+
+		$blocks_metadata = static::get_blocks_metadata();
+		$style_nodes     = static::get_style_nodes(
+			$incoming_data,
+			$blocks_metadata,
+			array(
+				'include_block_style_variations' => true,
+			)
+		);
+		foreach ( $style_nodes as $style_node ) {
+			$path = $style_node['path'];
+			/*
+			 * Background image styles should be replaced, not merged,
+			 * as they themselves are specific object definitions for the style.
+			 */
+			$background_image_path = array_merge( $path, static::PROPERTIES_METADATA['background-image'] );
```

Comment:
@aaronrobertshaw I need your smarts here. 

On the frontend end, the objects are being replaced fine _mostly_.

There's a final merge coming from `gutenberg_enqueue_global_styles` for which this line borks the operation:

https://github.com/WordPress/gutenberg/blob/5a399c1ae2c50b7a7a6daaf06c6dae3f4a264e30/lib/script-loader.php#L43-L43

I ran out of time to dig deeper, but commenting out that filter makes things work.

I noticed you removed that filter in some [block variation operations](https://github.com/WordPress/gutenberg/blob/5a399c1ae2c50b7a7a6daaf06c6dae3f4a264e30/lib/block-supports/block-style-variations.php#L163-L163).

Reply:
I'll take a look 👍 

From memory, removing those filters was to ensure the relevant block type data was still within the overall theme json data set. That way the block style variations support could retrieve the appropriate variation style values regardless of whether block assets were loading separately or not.

Reply:
At a quick glance, if we're expecting to work with the full unfiltered set of style nodes here, we might need to toggle that filter on and off like the block style variations support.

Another thing to double-check will be if there's a Gutenberg and Core version of that filter. Both would need handling.

Reply:
Thanks for looking. 

I forgot to add, the way to test is:

- using the test theme.json in the PR description,
- add a custom image for the quote block in global styles. 
- check the frontend. 
- the core/quote block should display your custom image, but it displays the one defined in the theme.json ref
- the reason is, as far as I can tell, it's not running through the new `merge()` changes

It could be something buggy in the code - I'll keep testing too.

Reply:
I think I might have bypassed it in [38b0e1d](https://github.com/WordPress/gutenberg/pull/64128/commits/38b0e1d8473c84411eb82072f21cfd63283ba910) by using `get_styles_block_nodes`, which bypasses the filter.

Reply:
Nice option. I clearly wasn't appreciating that we were working internally to `WP_Theme_JSON` here 🤦

Reply:
After following the test instructions above, this looks to be working well for me.